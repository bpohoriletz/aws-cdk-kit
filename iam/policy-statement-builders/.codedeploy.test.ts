import CodedeployBPolicyStatementBuilder from './codedeploy';

describe('new CodedeployBPolicyStatementBuilder', () => {
  let instance: CodedeployBPolicyStatementBuilder;

  beforeEach(() => {
    instance = new CodedeployBPolicyStatementBuilder();
  });

  test('has correct defaults set', () => {
    expect(instance).toMatchSnapshot();
  });

  test('.setFullAccess allows full access to CodeDeploy', () => {
    instance.setFullAccess();

    expect(instance).toMatchSnapshot();
  });

  test('.addResource adds resource', () => {
    instance.addResource('*');

    expect(instance).toMatchSnapshot();
  });
});
