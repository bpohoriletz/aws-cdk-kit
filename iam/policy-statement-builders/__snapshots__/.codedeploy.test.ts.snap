// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new CodedeployBPolicyStatementBuilder .addResource adds resource 1`] = `
CodedeployPolicyStatementBuilder {
  "policyStatementProps": PolicyStatementProduct {
    "actions": [],
    "resources": [
      "*",
    ],
  },
}
`;

exports[`new CodedeployBPolicyStatementBuilder .setFullAccess allows full access to CodeDeploy 1`] = `
CodedeployPolicyStatementBuilder {
  "policyStatementProps": PolicyStatementProduct {
    "actions": [
      "codedeploy:*",
    ],
    "resources": [],
  },
}
`;

exports[`new CodedeployBPolicyStatementBuilder has correct defaults set 1`] = `
CodedeployPolicyStatementBuilder {
  "policyStatementProps": PolicyStatementProduct {
    "actions": [],
    "resources": [],
  },
}
`;
